graphql_compiler/macros/macro_expansion.py

```diff
@@ -331,7 +331,12 @@ class name -> (macro edge name -> MacroEdgeDescriptor)
                     macro_registry, inherited_macro_edges, vertex_field_type,
                     new_selection_ast, new_query_args, tag_names)
 
-        if new_selection_ast is not selection_ast or selections_before or selections_after:
+        if new_selection_ast is selection_ast and (selections_before or selections_after):
+            raise AssertionError(u'No macro expansion happened but there are selections '
```

Comment:
Do you think we could somehow work in the macro we thought we were expanding here, and what field we were in / which field in there we were expanding? Otherwise this would be pretty hard to debug, if it ever happened.